backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -448,51 +538,65 @@ private static void checkSystemEndpoint(EndpointType endpointType) {
         }
     }
 
-    private String setupOpenBridgeProcessor(Endpoint endpoint, CamelProperties properties, String processorName) {
-
-        Map<String, Object> out = new HashMap<>();
-        out.put("name", processorName);
 
-        List<Object> filters = new ArrayList<>();
-        out.put("filters", filters);
-        filters.add(mkFilter("StringEquals", "rhaccount", endpoint.getAccountId()));
-        filters.add(mkFilter("StringEquals", OB_PROCESSOR_NAME, processorName));
+    /*
+     * Creates a template processor object from the provided
+     * endpoint.
+     */
+    private Processor createProcessor(Endpoint endpoint) {
 
-        Map<String, Object> actionMap = new HashMap<>();
-        out.put("action", actionMap);
+        CamelProperties properties = endpoint.getProperties(CamelProperties.class);
+        String processorName = properties.getExtras().get(OB_PROCESSOR_NAME);
+        String processorId = properties.getExtras().get(OB_PROCESSOR_ID);
 
-        actionMap.put("type", SLACK); // needs to be literally like this to the Slack Action
+        Processor out = new Processor(processorName);
 
-        // A qute template we should get it from DB
-        String template = "Hello from *Notifications* via _OpenBridge_ with {data.events.size()} event{#if data.events.size() > 1}s{/} from Application _{data.application}_ in Bundle _{data.bundle}_\n" +
-                "Events: {data.events} \n" +
-                "{#if data.context.size() > 0} Context is:\n" +
-                "{#each data.context}*{it.key}* -> _{it.value}_\n" +
-                "{/each}{/if}\n" +
-                "Brought to you by :redhat:\n";
-        out.put("transformationTemplate", template);
-
-        Map<String, String> props = new HashMap<>();
-        props.put("slack_channel", properties.getExtras().getOrDefault("channel", "#general"));
-        props.put("slack_webhook_url", properties.getUrl());
-        actionMap.put("parameters", props);
+        if (processorId != null) {
+            out.setId(processorId);
+        }
 
-        String token = bridgeAuth.getToken();
+        List<Processor.Filter> filters = out.getFilters();
+        filters.add(mkFilter("StringEquals", "rhaccount", endpoint.getAccountId()));
+        filters.add(mkFilter("StringEquals", OB_PROCESSOR_NAME, processorName));
 
-        Map<String, Object> pInfo = bridgeApiService.addProcessor(bridge.getId(), token, out);
+        Processor.Action action = new Processor.Action(SLACK_ACTION);
+        Map<String, String> props = action.getParameters();
+        props.put(SLACK_CHANNEL, properties.getExtras().getOrDefault(SLACK_CHANNEL, "#general"));
```

Comment:
Should the default channel be configurable?

Reply:
A fresh slack instance always comes with #random and #general   I don't think picking either on notifcations global level via config entry makes a change to hardcoding it here. And per customer seems like overkill to me. 
We could/should perhaps add some validation rules in out API that for slack the channel always needs to be provided. Likewise for other upcoming types.

Reply:
When I wrote "configurable", I was thinking of something coming from the DB as it would indeed make no sense to configure that from `application.properties`.

Enforcing a non null/empty Slack channel sounds more reasonable 👍